pkg/cgroups/subsystem.go

```diff
@@ -0,0 +1,58 @@
+package cgroups
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+// ProcSubsystemer returns cgroup mountpoints based on the /proc filesystem
+type ProcSubsystemer struct {
+	CachedSubsystems *Subsystems
+}
+
+var DefaultSubsystemer = &ProcSubsystemer{}
+
+func (ps *ProcSubsystemer) Find() (Subsystems, error) {
+	// For details about how this file is structured, refer to `man proc` or
+	// https://www.kernel.org/doc/Documentation/filesystems/proc.txt section 3.5
+	mountInfo, err := os.Open("/proc/self/mountinfo")
+	if err != nil {
+		return Subsystems{}, err
+	}
+	defer mountInfo.Close()
+	if ps.CachedSubsystems != nil {
```

Comment:
nit: you can do this before opening the proc file no? or are you trying to confirm that it is still there